src/Application/Mapping.php

```diff
@@ -15,6 +15,10 @@ public function __construct(
 		if ( !str_contains( $this->predicate, ':' ) ) {
 			throw new InvalidArgumentException( 'Invalid predicate' );
 		}
+
+		if ( !str_contains( $this->object, '://' ) ) {
```

Comment:
I replaced the filter check with this simpler check. Getting the full validation in here is too difficult and it's handled by the save use case anyway.